sdks/paths.mk

```diff
@@ -1,40 +1,49 @@
 -include $(TOP)/sdks/Make.config
 
-#Default paths
-
-SDK_DIR = $(abspath $(HOME)/Library/Developer/Xamarin/android-sdk-macosx/)
-NDK_DIR = $(abspath $(HOME)/Library/Developer/Xamarin/android-ndk/android-ndk-r14b)
-
-XCODE_DIR = /Applications/Xcode.app/Contents/Developer
-
-#Probe for alternative paths
+ifndef DISABLE_ANDROID
 
-#On some systems the sdk is named mac_x86
-ifeq ($(and $(wildcard $(SDK_DIR)/tools),$(wildcard $(SDK_DIR)/platform-tools)),)
-SDK_DIR = $(abspath $(HOME)/Library/Developer/Xamarin/android-sdk-mac_x86/)
+CheckSdkDir=$(or $(and $(wildcard $(1)/tools),$(wildcard $(1)/platform-tools)),$(warning Could not find Android SDK in "$(1)"))
+
+ifneq ($(call CheckSdkDir,$(HOME)/android-toolchain/sdk),)
+SDK_DIR = $(abspath $(HOME)/android-toolchain/sdk)
+else
+ifneq ($(call CheckSdkDir,$(HOME)/Library/Developer/Xamarin/android-sdk-macosx),)
+SDK_DIR = $(abspath $(HOME)/Library/Developer/Xamarin/android-sdk-macosx)
+else
+ifneq ($(call CheckSdkDir,$(HOME)/Library/Developer/Xamarin/android-sdk-mac_x86),)
+SDK_DIR = $(abspath $(HOME)/Library/Developer/Xamarin/android-sdk-mac_x86)
+else
+$(error Could not find Android SDK)
 endif
-
-#Latest XS installs NDK in this PATH
-ifeq ($(wildcard $(NDK_DIR)/platforms),)
-NDK_DIR = $(abspath $(HOME)/Library/Developer/Xamarin/android-sdk-mac_x86/ndk-bundle)
 endif
-
-#Error if tools are not found
-
-ifndef DISABLE_ANDROID
-ifeq ($(and $(wildcard $(SDK_DIR)/tools),$(wildcard $(SDK_DIR)/platform-tools)),)
-$(error Could not find Android SDK in $(SDK_DIR))
 endif
 
-ifeq ($(wildcard $(NDK_DIR)/platforms),)
-$(error Could not find Android NDK in $(NDK_DIR))
+CheckNdkDir=$(or $(and $(wildcard $(1)/platforms)),$(warning Could not find Android NDK in "$(1)"))
+
+ifneq ($(call CheckNdkDir,$(HOME)/android-toolchain/ndk),)
+NDK_DIR=$(abspath $(HOME)/android-toolchain/ndk)
+else
+ifneq ($(call CheckNdkDir,$(HOME)/Library/Developer/Xamarin/android-ndk/android-ndk-r14b),)
+NDK_DIR=$(abspath $(HOME)/Library/Developer/Xamarin/android-ndk/android-ndk-r14b)
+else
+ifneq ($(call CheckNdkDir,$(HOME)/Library/Developer/Xamarin/android-sdk-mac_x86/ndk-bundle),)
+NDK_DIR=$(abspath $(HOME)/Library/Developer/Xamarin/android-sdk-mac_x86/ndk-bundle)
+else
+$(error Could not find Android NDK)
+endif
 endif
-
 endif
 
+endif # DISABLE_ANDROID
+
 ifndef DISABLE_IOS
-ifeq ($(wildcard $(XCODE_DIR)),)
-$(error Could not find XCode in $(XCODE_DIR))
+
+CheckXcodeDir=$(or $(and $(wildcard $(1))),$(warning Could not find Xcode in "$(1)"))
+
+ifneq ($(call CheckXcodeDir,/Applications/Xcode.app/Contents/Developer),)
+XCODE_DIR=/Applications/Xcode.app/Contents/Developer
```

Comment:
`XCODE_DIR` should be the output of `xcode-select --print-path`

Reply:
Good point, I will update that